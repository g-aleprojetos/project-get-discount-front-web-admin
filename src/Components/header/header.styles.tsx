import styled from 'styled-components';
import colors from 'resources/colors';

export const ContainerContent = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  position: absolute;
  width: 100%;
  height: 34px;
  top: 55px;
`;

export const ContainerHeader = styled.div`
  display: flex;
  width: 100%;
  height: 144px;
  background-color: ${colors.lightgray};
`;

export const ContainerLogout = styled.div`
  position: absolute;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
  background-color: ${colors.black50per};
`;
