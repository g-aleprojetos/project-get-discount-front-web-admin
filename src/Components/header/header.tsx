import React, {useState} from 'react';
import {ILoginResponse} from 'resources/interfaces';
import {useNavigate} from 'react-router-dom';
import rotas from 'resources/rotas';
import {Popup} from 'Components/popup';
import colors from 'resources/colors';
import {Nav} from 'Components/nav/nav';
import {useAuthenticatorContext} from 'context/authetication';
import * as S from './header.styles';

export const Header = () => {
  const navigate = useNavigate();
  const {handleAutenticado} = useAuthenticatorContext();
  const [openLogout, setOpenLogout] = useState<boolean>(false);
  const handleLogout = () => {
    setOpenLogout(prev => !prev);
  };

  const handleOnConfirmarLogout = () => {
    handleLogout();
    navigate(rotas.Login);
    localStorage.clear();
    const logOut: ILoginResponse = {
      auth: false,
      accessToken: '',
      refreshToken: {id: '', expiresIn: 0, usuarioId: ''},
    };
    handleAutenticado(logOut);
  };

  return (
    <>
      <S.ContainerHeader data-testid={'test_header'}>
        <S.ContainerContent>
          <Nav handleLogout={handleLogout} />
        </S.ContainerContent>
      </S.ContainerHeader>
      {openLogout && (
        <Popup
          textTitle="Deseja fazer Logout?"
          aoSair={handleLogout}
          botaoConfirmar={{
            legenda: 'Confirmar',
            corTexto: colors.white,
            corBotao: colors.caribbeanGreen,
            aoClicar: () => {
              handleOnConfirmarLogout();
            },
          }}
          botaoCancelar={{
            legenda: 'Cancelar',
            corTexto: colors.white,
          }}></Popup>
      )}
    </>
  );
};
