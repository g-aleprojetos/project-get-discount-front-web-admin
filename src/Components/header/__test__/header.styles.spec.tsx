import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import * as S from '../header.styles';

describe('Input.styles', () => {
  test('ContainerContent DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.ContainerContent />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('ContainerHeader DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.ContainerHeader />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('ContainerLogout DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.ContainerLogout />).toJSON();

    expect(container).toMatchSnapshot();
  });
});
