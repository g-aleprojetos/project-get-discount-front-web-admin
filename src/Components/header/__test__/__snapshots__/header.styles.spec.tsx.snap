// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input.styles ContainerContent DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: absolute;
  width: 100%;
  height: 34px;
  top: 55px;
}

<div
  className="c0"
/>
`;

exports[`Input.styles ContainerHeader DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 144px;
  background-color: #D3D3d3;
}

<div
  className="c0"
/>
`;

exports[`Input.styles ContainerLogout DEVE ser igual ao snapshot 1`] = `
.c0 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
  background-color: #00000080;
}

<div
  className="c0"
/>
`;
